zuul/connection/github.py

```diff
@@ -366,6 +367,25 @@ def _init_data(self):
         return data
 
 
+class DropMaxAgeHeaders(BaseHeuristic):
+    """CacheControl will prioritize the max-age headers from our github
+    responses over the ETag, causing all caching to be time-based and only
+    refresh every 60 seconds.  Stripping them out here forces the library
+    to use the ETag headers and a conditional request when determining
+    freshness of existing cache entries.
+    """
+    def update_headers(self, response):
+        cc_header = response.headers.get('cache-control')
+        if not cc_header:
+            return {}
+        cc_new_header = []
+        for cc in cc_header.split(','):
+            cc = cc.strip()
+            if not cc.startswith('max-age') and not cc.startswith('s-maxage'):
+                cc_new_header.append(cc)
```

Comment:
Feels like we could do something with a list comprehension, or  a filter with a lambda, but that's just trying to be clever. This works too!